pointcloud/basic_octree.go

```diff
@@ -96,7 +98,7 @@ func (octree *basicOctree) At(x, y, z float64) (pc.Data, bool) {
 // to either all the data points or a subset of them based on the given numBatches and currentBatch
 // inputs. If any of the applied functions returns a false value, iteration will stop and no further
 // points will be processed.
-func (octree *basicOctree) Iterate(numBatches, currentBatch int, fn func(p r3.Vector, d pc.Data) bool) {
+func (octree *BasicOctree) Iterate(numBatches, currentBatch int, fn func(p r3.Vector, d Data) bool) {
```

Comment:
I know this isn't changing any behavior, but just for my own edification:

What are the properties / semantics a given `Iterate` method must satisfy, on a pointcloud?

 Example: Is traversing the PointCloud in any order allowed or is there a certain ordering guarantee the caller will expect as the Iterate callback function traverses the PointCloud or are the guarantees more similar to `any deterministic traversal order for a given pointcloud is allowed, as long as no points are visited twice`?

Reply:
The second one is correct. There is no defined order requirement. For a typical pointcloud, which is a list, the order of traversal is obvious. For an octree it could be depth or breath first. However as you stated deterministic nature is required to allow for repeatable batching processes